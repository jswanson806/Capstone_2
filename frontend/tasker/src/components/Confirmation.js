import React from "react";

const Confirmation = ({targetUserId, job, onConfirm, onClose}) => {

    const handleConfirmation = () => {
        console.log(targetUserId)
        onConfirm(job.id, targetUserId);
        onClose();
    }


    return (
        <div>
            <h3>Assign user to job?</h3>
            <button onClick={handleConfirmation}>Confirm</button>
            <button onClick={onClose}>Cancel</button>
        </div>
    )
};

export default Confirmation;