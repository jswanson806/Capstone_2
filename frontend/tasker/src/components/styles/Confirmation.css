.confirmation-container {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 2em 0 2em;
}
    
.confirmation-centered-card {
    text-align: center;
    max-height: 700px;
    width: 100%;
    margin: 0 auto;
    background-color: #fff;
    border-radius: 20px;
    padding: 20px;
    transition: max-height 0.2s ease-in-out;
}

.button {
    width: 10em;
}