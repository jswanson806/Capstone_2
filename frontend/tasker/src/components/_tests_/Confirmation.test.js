import React from 'react';
import {fireEvent, render} from '@testing-library/react';
import Confirmation from '../../components/Confirmation.js';
import TaskerApi from '../../api.js';

const onClose = jest.fn();
const onConfirm = jest.fn();
const updateSingleJob = jest.spyOn(TaskerApi, 'updateSingleJob')

const job = 
        {
            title: 'test title',
            body: 'test body',
            address: 'test address',
            assigned_to: '',
            status: 'pending',
            posted_by: 1,
            before_image_url: 'formData.beforeImage'
        };

describe("createJob form smoke and snapshot tests", () => {

    it("should render without crashing", async () => {

        render(
            <Confirmation targetUserId={1} job={job} onConfirm={onConfirm} onClose={onClose}/>
        );

    });

    it("should match snapshot", async() => {
        let asFragment;

        const {asFragment: fragment} = render(
            <Confirmation targetUserId={1} job={job} onConfirm={onConfirm} onClose={onClose}/>
        );

        asFragment = fragment;

        expect(asFragment()).toMatchSnapshot();
    })
})

describe("createJob form renders correct content", () => {


    it("should match snapshot", async() => {

        const {getByTestId} = render(
            <Confirmation targetUserId={1} job={job} onConfirm={onConfirm} onClose={onClose}/>
        );

        expect(getByTestId("confirmation-confirm-button")).toBeInTheDocument();
        expect(getByTestId("confirmation-cancel-button")).toBeInTheDocument();
    })
})

describe("confirm button calls onConfirm function, which calls assignToJob and updateSingleJob", () => {

    it("should match snapshot", async() => {

        const {getByTestId} = render(
            <Confirmation targetUserId={1} job={job} onConfirm={onConfirm} onClose={onClose}/>
        );
        
        const confirmBtn = getByTestId("confirmation-confirm-button")
        expect(confirmBtn).toBeInTheDocument();
   
        fireEvent.click(confirmBtn);

        expect(onConfirm).toHaveBeenCalled();
    })

})