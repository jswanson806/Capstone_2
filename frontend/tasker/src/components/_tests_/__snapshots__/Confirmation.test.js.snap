// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createJob form smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="confirmation-container"
  >
    <div
      class="confirmation-centered-card"
    >
      <div
        class="modal-header"
      >
        <h5
          class="modal-title"
        >
          Are you sure you want to assign this applicant to your job?
        </h5>
      </div>
      <div
        class="modal-body"
      >
        <button
          class="button btn btn-success"
          data-testid="confirmation-confirm-button"
          type="button"
        >
          Confirm
        </button>
        <button
          class="button btn btn-danger"
          data-testid="confirmation-cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
